### PR TITLE
Limit Parser Recursion

### PR DESCRIPTION
**Pull request by jaymell**
Eliminate possibility of stack overflow when
parsing server responses by limiting recursion 
depth.

### CODE REVIEW

**Comment by jaymell** [comment]
This could probably be more declarative/elegant, but this seems to at least meet the basic need.

File: redis/src/parser.rs [hunk 1]
```diff
@@ -174,7 +186,7 @@ mod aio_support {
                 let buffer = &bytes[..];
                 let mut stream =
                     combine::easy::Stream(combine::stream::MaybePartialStream(buffer, !eof));
-                match combine::stream::decode_tokio(value(), &mut stream, &mut self.state) {
+                match combine::stream::decode_tokio(value(None), &mut stream, &mut self.state) {
```
Comment (djc) [inline redis/src/parser.rs:191]: Just a question: I notice that there is a `decode_tokio()`. Is there another variant for other runtimes that we're not compiling in the default feature set/in CI?

**Review by djc — APPROVED** [review]
Seems fine to me!

File: redis/src/parser.rs (hunk 1 above)
Comment (jaymell) [reply redis/src/parser.rs:191]: I'm not sure but it does look like `combine` may have an alternative decode macro for `async-std`.

**Pull request merged**